float32 x
float32 y
float32 z
float32 orientation
time timestamp
uint32 vector_len
gate[] vector_gates
